From ef5ba6e4297c56b98613993dbc28321ff914ec1f Mon Sep 17 00:00:00 2001
From: Jake Hamilton <jakehamtexas@gmail.com>
Date: Sun, 11 Oct 2026 00:19:51 -0500
Subject: [PATCH] Widen support for kernel versions without drvwrap

On mainline kernels, drvwrap was removed in 6.8. The change was
backported to stable 6.1.188 and 6.6.157, as well.
---
 os_dep/linux/usb_intf.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/os_dep/linux/usb_intf.c b/os_dep/linux/usb_intf.c
index ccf42ba..73e85e7 100644
--- a/os_dep/linux/usb_intf.c
+++ b/os_dep/linux/usb_intf.c
@@ -245,7 +245,9 @@ struct rtw_usb_drv usb_drv = {
 	.usbdrv.reset_resume   = rtw_dev_resume,
 #endif
 
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 8, 0)) || (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 19))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 6, 157)) || \
+    (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 188) && LINUX_VERSION_CODE < KERNEL_VERSION(6, 2, 0)) || \
+    (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 19))
 	.usbdrv.driver.shutdown = rtw_dev_shutdown,
 #else
 	.usbdrv.drvwrap.driver.shutdown = rtw_dev_shutdown,
-- 
2.55.0

